server/api/crud/runtimes/nuclio/function.py

```diff
@@ -486,6 +496,21 @@ def _set_function_scheduling_params(function, nuclio_spec):
             )
 
 
+def _enrich_function_node_selector_with_project(
+    db_session, project_name, function_node_selector
+):
+    if db_session and project_name:
+        project = server.api.utils.singletons.db.get_db().get_project(
+            session=db_session, name=project_name
+        )
+        if project.spec.default_function_node_selector:
+            return mlrun.utils.helpers.merge_with_precedence(
+                project.spec.default_function_node_selector,
+                function_node_selector,
+            )
+    return function_node_selector
```

Comment:
uncertain why it has been decided to carry the db_session all the way here and get the project from db while
server launcher enriches the runtime for us @ https://github.com/mlrun/mlrun/blob/5883a9ea1d4dc8035ca6112cfc21a556f285b8a3/server/api/launcher.py#L213

you might have wanted to enrich the function config before sending it to nuclio, right? but the outcome is that you have nuclio CRD with the node selector and not the mlrun-function itself so I think this is yet to accomplish the desire of "export/import" project without making it dirty with per-ssytem node selectors.

perhaps what we can do here is, instead of carry the db session to get the project, rather pass the project object that we have already resolved. wdyt?

Reply:
If I understand correctly, the launcher flow doesn't apply to the deploying of Nuclio functions, which is why I chose to handle it this way. Regarding your suggestion to pass the project object instead of the db_session, I agree it's cleaner. However, could you clarify how this change would specifically address the import/export issue? not sure I got what you meant there

Reply:
To document here as well after follow up discussion - this change creates a discrepancy between the behavior of nuclio functions and mlrun functions. The nuclio function will be affected by per-system node selectors, making it "dirty," while the mlrun function will not. Typically, we let mlrun dictate the behavior of its runtimes, but in this case, we have decided to let nuclio handle the merge logic on it's own. Consequently, we will remove all extra behavior related to nuclio functions from mlrun in this context.